llms/mlx_lm/server.py

```diff
@@ -544,6 +564,11 @@ def main():
     model, tokenizer = load(
         args.model, adapter_path=args.adapter_path, tokenizer_config=tokenizer_config
     )
+
+    if args.use_default_chat_template:
+        if tokenizer.chat_template is None:
+            tokenizer.chat_template = tokenizer.default_chat_template
```

Comment:
What's the purpose of that?

Reply:
DBRX (https://huggingface.co/mlx-community/dbrx-instruct-4bit) needs an explicit default chat template. This is how it's done in generate.py. See also https://huggingface.co/mlx-community/dbrx-instruct-4bit/discussions/7

Reply:
Right, sounds good, thanks!